src/plugins/intel_npu/src/plugin/npuw/partitioning/online/snapshot.cpp

```diff
@@ -45,11 +45,35 @@ bool isOp(const std::shared_ptr<ov::Node>& node) {
     }
     return true;
 }
+
+std::vector<ov::element::Type> getWeightsPrecision(const std::shared_ptr<ov::Node>& node) {
+    NPUW_ASSERT(!ov::op::util::is_constant(node) && !ov::op::util::is_parameter(node) &&
+                !ov::op::util::is_output(node));
+
+    std::vector<ov::element::Type> precisions;
+
+    for (size_t i = 0; i < node->inputs().size(); ++i) {
+        auto target_input = node->get_input_source_output(i);
+        auto ov_node_parent = target_input.get_node()->shared_from_this();
+
+        if (ov::is_type<ov::opset1::Convert>(ov_node_parent) && ov_node_parent->inputs().size() == 1) {
```

Comment:
If the node is Convert, it always has just one input. Did you want to check the number of its producers here?

Reply:
Gotcha, removed the second check